File: deaf_grandma.py
def deaf_grandma():
    bye = 0
    greeting = input('HEY KID!\n')
    while not greeting:
        greeting = input('WHAT?!\n')
    while greeting:
        if greeting == greeting.upper() and not greeting == 'GOODBYE':
            greeting = input('NO, NOT SINCE 1946!\n')
        elif greeting == greeting.lower():
            greeting = input('SPEAK UP, KID!\n')
        elif greeting == 'GOODBYE' and bye < 1:
            greeting = input('LEAVING SO SOON?\n')
            bye+=1
        elif greeting == 'GOODBYE':
            print('LATER, SKATER!')
            return 0
        while not greeting:
            greeting = input('WHAT?!\n')
deaf_grandma() 

File: deaf_grandma.js
function deafGrandma() {
    let bye = 0
    let text = prompt('HEY KID!')
    while (bye < 2) {
        if (text.toLowerCase() === 'goodbye!' && bye === 1) {
            text = alert('LATER, SKATER!')
            break
        } else if (text.toLowerCase() === 'goodbye!') {
            bye++ 
            text = prompt('LEAVING SO SOON?')
        } else if (text === "") {
            text = prompt('WHAT?!')
        } else if (text === text.toLowerCase()) {
            text = prompt('SPEAK UP, KID!')
        } else if (text === text.toUpperCase()) {
            text = prompt('NO, NOT SINCE 1946!')
        }  
    }
}
deafGrandma()